### PR TITLE
fix: make walk max retries parameter configurable

### PR DESCRIPTION
**Pull request by omrozowicz-splunk**
# Description

This PR is to make walk max retries configurable, as this is hardcoded to be 50 retries and it takes a lot of resources and time to walk devices that are unavailable.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Refactor/improvement
- [x] This change requires a documentation update

## How Has This Been Tested?

Unit tests

## Checklist

- [x] My commit message is [conventional](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] I have run pre-commit on all files before creating the PR
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: rendered/manifests/tests/splunk-connect-for-snmp/templates/inventory/job.yaml [hunk 1]
```diff
@@ -44,6 +44,8 @@ spec:
             value: "http://release-name-mibserver/standard.txt"
           - name: LOG_LEVEL
             value: INFO
+          - name: CHAIN_OF_TASKS_EXPIRY_TIME
+            value: "60"
```
Comment (ajasnosz) [inline rendered/manifests/tests/splunk-connect-for-snmp/templates/inventory/job.yaml:48]: Is this variable related to walk max retry?

**Review by ajasnosz — APPROVED** [review]

File: rendered/manifests/tests/splunk-connect-for-snmp/templates/inventory/job.yaml (hunk 1 above)
Comment (omrozowicz-splunk) [reply rendered/manifests/tests/splunk-connect-for-snmp/templates/inventory/job.yaml:48]: no, this is from this PR: https://github.com/splunk/splunk-connect-for-snmp/pull/838 , no `make render` was executed there

**Review by wojtekzyla — APPROVED** [review]
